### PR TITLE
Migrate type tests to JUnit

### PR DESCRIPTION
**Pull request by martint**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Review by wendigo — APPROVED** [review]

File: testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java [hunk 1]
```diff
@@ -190,6 +190,10 @@ private static boolean isTestAnnotated(Method method)
                         // testng annotation (@Test, @Before*, @DataProvider, etc.)
                         return true;
                     }
+                    if (isJUnitAnnotation(annotationClass)) {
```
Comment (findepi) [inline testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java:193]: the stated goal of this change is

> This makes it possible to migrate tests gradually by allowing
test subclasses to have a mix of TestNG and JUnit tests
when some of them are inherited.

but i don't believe this works.
i don't see how a mix of TestNG and JUnit tests would actually work together with setup methods.

is this only for stateless tests that do not have setup methods?
This seem to specifically remove any safety guarantees that we had for test inheritance tree.

Comment (findepi) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java:193]: cc @ebyhr @hashhar @losipiuk please correct me if i am wrong

Comment (martint) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java:193]: > but i don't believe this works.

It works fine. Why wouldn't it? Each testing framework only looks at the annotations it cares about. See https://github.com/martint/tests:

```
> mvn test

[...]

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.weakref.tests.JUnitTests
Setup: org.weakref.tests.JUnitTests
JUnit test
Teardown: org.weakref.tests.JUnitTests
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.031 s -- in org.weakref.tests.JUnitTests
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running TestSuite
Setup: org.weakref.tests.TestNgTests
TestNG test
Teardown: org.weakref.tests.TestNgTests
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.184 s -- in TestSuite
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO]
```

Comment (findepi) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java:193]: i don't think i understand, maybe i just cannot think abstract enough. can you walk me by example in Trino context, why this works and why it's a good change?

Comment (martint) [reply testing/trino-testing-services/src/main/java/io/trino/testng/services/ReportBadTestAnnotations.java:193]: Before that change, if a base class is annotated with any TestNG annotation, those checkers fail if a subclass has methods that are not annotated with TestNG's `@Test` annotation. If you want to write a test class that extends from that base class but use JUnit instead, the checker doesn't consider it a valid TestNG test and fails.

Without such a change, it's extremely hard to migrate tests incrementally. It requires migrating *all* tests that derive from a given TestNG-annotated base class to JUnit in one shot, or to duplicate the base class w/o TestNG annotations.

**Pull request merged**